Component flow not working

For following `ComponentFlow`:

```java
List<SelectItem> names = List.of(
        SelectItem.of("Name 1", "n1"),
        SelectItem.of("Name 2", "n2"),
        SelectItem.of("Name 3", "n3"),
        SelectItem.of("Name 4", "n4")
);

ComponentFlow componentFlow = componentFlowBuilder.clone().reset()
        .withStringInput("login")
        .name("Login:")
        .defaultValue("j.doe")
        .and()

        .withStringInput("password")
        .name("Password:")
        .maskCharacter('*')
        .and()

        .withNumberInput("id")
        .name("User ID:")
        .required()
        .and()

        .withMultiItemSelector("multiple-names")
        .name("Names:")
        .max(2)
        .selectItems(names)
        .and()

        .withSingleItemSelector("single-name")
        .name("Name:")
        .and()

        .withPathInput("file")
        .name("File path:")
        .and()

        .withConfirmationInput("confirm")
        .name("Confirm:")
        .and()

        .build();

ComponentFlow.ComponentFlowResult flowResult = componentFlow.run();
ComponentContext<?> context = flowResult.getContext();
```

We are getting errors:

```
shell:>flow
org.springframework.shell.core.command.CommandExecutionException: Unable to execute command flow
	at org.springframework.shell.core.command.CommandExecutor.execute(CommandExecutor.java:74)
	at org.springframework.shell.core.InteractiveShellRunner.run(InteractiveShellRunner.java:99)
	at org.springframework.shell.core.autoconfigure.ShellRunnerAutoConfiguration.lambda$springShellApplicationRunner$0(ShellRunnerAutoConfiguration.java:41)
	at org.springframework.boot.SpringApplication.lambda$callRunner$0(SpringApplication.java:788)
	at org.springframework.util.function.ThrowingConsumer$1.acceptWithException(ThrowingConsumer.java:82)
	at org.springframework.util.function.ThrowingConsumer.accept(ThrowingConsumer.java:60)
	at org.springframework.util.function.ThrowingConsumer$1.accept(ThrowingConsumer.java:86)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:800)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:788)
	at org.springframework.boot.SpringApplication.lambda$callRunners$0(SpringApplication.java:776)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:186)
	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:571)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:153)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:176)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:632)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:776)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:328)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1365)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354)
	at com.example.ComponentsApplication.main(ComponentsApplication.java:9)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:106)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:40)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.springframework.util.MethodInvoker.invoke(MethodInvoker.java:275)
	at org.springframework.shell.core.command.adapter.MethodInvokerCommandAdapter.doExecute(MethodInvokerCommandAdapter.java:102)
	at org.springframework.shell.core.command.AbstractCommand.execute(AbstractCommand.java:151)
	at org.springframework.shell.core.command.CommandExecutor.execute(CommandExecutor.java:71)
	... 27 more
Caused by: java.io.UncheckedIOException: java.io.FileNotFoundException: class path resource [org/springframework/shell/component/string-input-default.stg] cannot be opened because it does not exist
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.resourceAsString(AbstractComponent.java:346)
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.renderTemplateResource(AbstractComponent.java:227)
	at org.springframework.shell.jline.tui.component.StringInput.access$000(StringInput.java:42)
	at org.springframework.shell.jline.tui.component.StringInput$DefaultRenderer.apply(StringInput.java:328)
	at org.springframework.shell.jline.tui.component.StringInput$DefaultRenderer.apply(StringInput.java:324)
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.render(AbstractComponent.java:133)
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.loop(AbstractComponent.java:295)
	at org.springframework.shell.jline.tui.component.support.AbstractTextComponent.runInternal(AbstractTextComponent.java:75)
	at org.springframework.shell.jline.tui.component.support.AbstractTextComponent.runInternal(AbstractTextComponent.java:41)
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.run(AbstractComponent.java:168)
	at org.springframework.shell.jline.tui.component.flow.ComponentFlow$DefaultComponentFlow.lambda$stringInputsStream$1(ComponentFlow.java:540)
	at org.springframework.shell.jline.tui.component.flow.ComponentFlow$DefaultComponentFlow.runGetResults(ComponentFlow.java:471)
	at org.springframework.shell.jline.tui.component.flow.ComponentFlow$DefaultComponentFlow.run(ComponentFlow.java:410)
	at com.example.ComponentsFlow.generate(ComponentsFlow.java:71)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 32 more
Caused by: java.io.FileNotFoundException: class path resource [org/springframework/shell/component/string-input-default.stg] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:212)
	at org.springframework.shell.jline.tui.component.support.AbstractComponent.resourceAsString(AbstractComponent.java:342)
	... 46 more
Unable to run command flow: null
```

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/confirmation-input-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/confirmation-input-default.stg
@@ -1,0 +1,40 @@
+// message
+message(model) ::= <%
+<if(model.message && model.hasMessageLevelError)>
+<({<figures.error>}); format="style-level-error"> <model.message; format="style-level-error">
+<elseif(model.message && model.hasMessageLevelWarn)>
+<({<figures.warning>}); format="style-level-warn"> <model.message; format="style-level-warn">
+<elseif(model.message && model.hasMessageLevelInfo)>
+<({<figures.info>}); format="style-level-info"> <model.message; format="style-level-info">
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <%
+<if(model.defaultValue)>
+  <("(Y/n)"); format="style-item-disabled">
+<else>
+  <("(y/N)"); format="style-item-disabled">
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.resultValue; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)> <model.input>
+<message(model)>
+>>
+
+// main
+main(model) ::= <<
+<if(model.resultValue)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/multi-item-selector-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/multi-item-selector-default.stg
@@ -1,0 +1,73 @@
+// selector rows
+truncate(name,model) ::= <%
+<name; format={truncate-width:<model.terminalWidth>-prefix:5}>
+%>
+
+// used to select style if item is selected/unselected
+selected_style(flag) ::= <%
+<if(flag)>style-item-selected<else>style-item-unselected<endif>
+%>
+
+// selector rows
+select_item(item,model) ::= <%
+<if(item.onrow)>
+  <({<figures.rightPointingQuotation> }); format="style-item-selector">
+<else>
+  <("  ")>
+<endif>
+
+<if(item.enabled)>
+  <if(item.selected)>
+    <({<figures.checkboxOn> }); format=selected_style(item.selected)> <truncate(item.name,model)>
+  <else>
+    <({<figures.checkboxOff> }); format=selected_style(item.selected)> <truncate(item.name,model)>
+  <endif>
+<else>
+  <if(item.selected)>
+     <({<figures.checkboxOn> }); format="style-item-disabled"> <({<truncate(item.name,model)>}); format="style-item-disabled">
+  <else>
+    <({<figures.checkboxOff> }); format="style-item-disabled"> <({<truncate(item.name,model)>}); format="style-item-disabled">
+  <endif>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// within info section, dedicated instructions for user
+info_filter(model) ::= <%
+<if(model.input)>
+, filtering '<model.input>'
+<else>
+, type to filter
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <<
+[Use arrows to move]<info_filter(model)>
+>>
+
+// get comma delited string
+comma_delimited(values) ::= <%
+<values; separator=",">
+%>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <(comma_delimited(model.values)); format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+<model.rows:{x|<select_item(x,model)>}; separator="\n">
+>>
+
+// main - hardcoded name
+// model - model built by MultiItemSelectorContext
+main(model) ::= <<
+<if(model.isResult)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/number-input-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/number-input-default.stg
@@ -1,0 +1,45 @@
+// message
+message(model) ::= <%
+<if(model.message && model.hasMessageLevelError)>
+  <({<figures.error>}); format="style-level-error"> <model.message; format="style-level-error">
+<elseif(model.message && model.hasMessageLevelWarn)>
+  <({<figures.warning>}); format="style-level-warn"> <model.message; format="style-level-warn">
+<elseif(model.message && model.hasMessageLevelInfo)>
+  <({<figures.info>}); format="style-level-info"> <model.message; format="style-level-info">
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <%
+<if(model.input)>
+  <model.input>
+<else>
+  <("[Number Type: "); format="style-value"><model.defaultClass; format="style-value"><("]"); format="style-value">
+  <if(model.defaultValue)>
+    <("[Default "); format="style-value"><model.defaultValue; format="style-value"><("]"); format="style-value">
+  <elseif(model.required)>
+    <("[Required]"); format="style-value">
+  <endif>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.resultValue; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+<message(model)>
+>>
+
+// main
+main(model) ::= <<
+<if(model.resultValue)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/path-input-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/path-input-default.stg
@@ -1,0 +1,38 @@
+// message
+message(model) ::= <%
+<if(model.message && model.hasMessageLevelError)>
+<({<figures.error>}); format="style-level-error"> <model.message; format="style-level-error">
+<elseif(model.message && model.hasMessageLevelWarn)>
+<({<figures.warning>}); format="style-level-warn"> <model.message; format="style-level-warn">
+<elseif(model.message && model.hasMessageLevelInfo)>
+<({<figures.info>}); format="style-level-info"> <model.message; format="style-level-info">
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <%
+<if(model.input)>
+<model.input>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.resultValue; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+<message(model)>
+>>
+
+// main
+main(model) ::= <<
+<if(model.resultValue)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/path-search-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/path-search-default.stg
@@ -1,0 +1,53 @@
+// message
+message(model) ::= <%
+<if(model.message && model.hasMessageLevelError)>
+<({<figures.error>}); format="style-level-error"> <model.message; format="style-level-error">
+<elseif(model.message && model.hasMessageLevelWarn)>
+<({<figures.warning>}); format="style-level-warn"> <model.message; format="style-level-warn">
+<elseif(model.message && model.hasMessageLevelInfo)>
+<({<figures.info>}); format="style-level-info"> <model.message; format="style-level-info">
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <%
+<if(model.input)>
+<model.input>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// render path item
+path_item(item,model) ::= <%
+<if(item.selected)>
+<({<figures.rightPointingQuotation> }); format="style-item-selector"><item.partsText; format={width:<model.terminalWidth>,prefix:2,textStyle:style-item-selector,matchStyle:style-level-warn}>
+<else>
+<("  ")><item.partsText; format={width:<model.terminalWidth>,prefix:2,textStyle:style-item-selector,matchStyle:style-level-warn}>
+<endif>
+%>
+
+// render paths
+paths(model) ::= <<
+<model.pathViewItems:{x|<path_item(x,model)>}; separator="\n">
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.resultValue; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+<message(model)>
+<paths(model)>
+>>
+
+// main
+main(model) ::= <<
+<if(model.resultValue)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/single-item-selector-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/single-item-selector-default.stg
@@ -1,0 +1,48 @@
+// selector rows
+truncate(name,model) ::= <%
+<name; format={truncate-width:<model.terminalWidth>-prefix:2}>
+%>
+
+select_item(item,model) ::= <%
+<if(item.selected)>
+<({<figures.rightPointingQuotation> }); format="style-item-selector"><({<truncate(item.name,model)>}); format="style-item-selector">
+<else>
+<("  ")><truncate(item.name,model)>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// within info section, dedicated instructions for user
+info_filter(model) ::= <%
+<if(model.input)>
+, filtering '<model.input>'
+<else>
+, type to filter
+<endif>
+%>
+
+// info section after '? xxx'
+info(model) ::= <<
+[Use arrows to move]<info_filter(model)>
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.value; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+<model.rows:{x|<select_item(x,model)>}; separator="\n">
+>>
+
+// main - hardcoded name
+// model - model built by SingleItemSelectorContext
+main(model) ::= <<
+<if(model.isResult)><result(model)><else><running(model)><endif>
+>>

--- a/spring-shell-jline/src/main/resources/org/springframework/shell/component/string-input-default.stg
+++ b/spring-shell-jline/src/main/resources/org/springframework/shell/component/string-input-default.stg
@@ -1,0 +1,40 @@
+// info section after '? xxx'
+info(model) ::= <%
+<if(model.hasMaskCharacter)>
+  <if(model.maskedInput)>
+    <model.maskedInput>
+  <else>
+    <if(model.defaultValue)>
+      <("[Default "); format="style-value"><model.defaultValue; format="style-value"><("]"); format="style-value">
+    <endif>
+  <endif>
+<else>
+  <if(model.input)>
+    <model.input>
+  <else>
+    <if(model.defaultValue)>
+      <("[Default "); format="style-value"><model.defaultValue; format="style-value"><("]"); format="style-value">
+    <endif>
+  <endif>
+<endif>
+%>
+
+// start '? xxx' shows both running and result
+question_name(model) ::= <<
+<({<figures.questionMark>}); format="style-list-value"> <model.name; format="style-title">
+>>
+
+// component result
+result(model) ::= <<
+<question_name(model)> <model.maskedResultValue; format="style-value">
+>>
+
+// component is running
+running(model) ::= <<
+<question_name(model)> <info(model)>
+>>
+
+// main
+main(model) ::= <<
+<if(model.resultValue)><result(model)><else><running(model)><endif>
+>>

